src/service/components/input.js

```diff
@@ -609,8 +609,12 @@ class Controller {
         try {
             this._ensureAdapter();
 
-            for (let i = 0; i < input.length; i++)
-                this._session.pressKey(input[i], modifiers);
+            if (typeof input === 'string') {
+                for (let i = 0; i < input.length; i++)
+                    this._session.pressKey(input[i], modifiers);
+            } else {
+                this._session.pressKey(input, modifiers);
+            }
```

Comment:
I'm a little wary of passing `input` to `pressKey()` without being more sure what type it _does_ have, if it's not a string. What other types would it be getting? Keysyms, I guess, which should be Numbers.

How do you feel about a slightly more explicit check?
```suggestion
            const in_type = typeof input;
            switch(in_type) {
                case 'number':
                    this._session.pressKey(input, modifiers);
                    break;
                case 'string':
                    for (let i = 0; i < input.length; i++)
                        this._session.pressKey(input[i], modifiers);
                    break;
                default:
                    throw new Error(`Unrecognized input type '${in_type}'`);
            }
```

Reply:
It shouldn't get any other type as input since `pressKeys` gets either `input.key` which should always be a string ([kde-connect android code](https://github.com/KDE/kdeconnect-android/blob/master/src/org/kde/kdeconnect/Plugins/MousePadPlugin/KeyListenerView.java#L85-L89)), or `input.specialKey` which is already checked to contain a valid keysym [here](https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/master/src/service/plugins/mousepad.js#L193).